File: python/examples/producer_demo.py
#!/usr/bin/env python3
"""Publish sample messages to a Redis stream."""

import os

from redis_streams_messaging import StreamConfig, StreamProducer


def main() -> None:
    config = StreamConfig(redis_url=os.getenv("REDIS_URL", "redis://localhost:6379/0"))
    producer = StreamProducer(config)
    stream = os.getenv("STREAM_NAME", "demo-events")
    message_id = producer.publish(stream, {"type": "demo", "value": 1})
    print(f"Published to {stream}: {message_id}")
    producer.close()


if __name__ == "__main__":
    main()

File: java/examples/ProducerDemo.java
package com.vipulconda.streams;

import java.util.Map;

public final class ProducerDemo {

    private ProducerDemo() {}

    public static void main(String[] args) throws Exception {
        String redisUrl = System.getenv().getOrDefault("REDIS_URL", "redis://localhost:6379");
        String stream = System.getenv().getOrDefault("STREAM_NAME", "demo-events");

        try (StreamProducer producer = new StreamProducer(new StreamConfig(redisUrl, "app-consumers", "producer-1", 3, 2000, 10))) {
            String id = producer.publish(stream, Map.of("type", "demo", "value", 1));
            System.out.println("Published to " + stream + ": " + id);
        }
    }
}
